bin_sort/bin_sort.c

```diff
@@ -0,0 +1,56 @@
+/* Following algorithm sorts the input array in ascending order
+ * Time Complexity : O(n)-> Avg case and O(n^2)-> Worst Case
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+struct bucket {
+	int count;
+	float* value;
+};
+
+int comparator(const float* first, const float* second) {
+	float a = *first, b = *second;
+	float result = a - b;
+	return (0 < result) - (result < 0);
+}
+
+void bucket_sort(int array[], int n) {
+	struct bucket buckets[3];
+	int a, b, c;
+	for (a = 0; a < 3; a++) {
```

Comment:
Making the bucket size 10 (or dynamic) will be a faster approach. Consider when the input has 10^6 numbers, having few buckets will be ineffective here.